src/mahoji/commands/clue.ts

```diff
@@ -155,6 +138,30 @@ export const clueCommand: OSBMahojiCommand = {
 		const hasOrnateJewelleryBox = poh.jewellery_box === getPOHObject('Ornate jewellery box').id;
 		const hasJewelleryBox = poh.jewellery_box !== null;
 		const hasXericTalisman = poh.amulet === getPOHObject("Mounted xeric's talisman").id;
+		const hasAchievementDiaryCape = user.hasEquippedOrInBank('Achievement diary cape');
+
+		// Stash Unit boost
+		const stashBoost = await getStashBoost(userID, clueTier.name);
+		boosts.push(`${stashBoost.toFixed(2)}% for built STASH Units`);
+		duration *= 1 - stashBoost / 100;
+
+		// Combat stats boost
+		if (['Hard', 'Elite', 'Master'].includes(clueTier.name)) {
+			const totalCombatStats =
+				user.skillLevel(SkillsEnum.Attack) +
+				user.skillLevel(SkillsEnum.Strength) +
+				user.skillLevel(SkillsEnum.Ranged);
+			let combatBoost = (totalCombatStats / (3 * 99)) * 100;
+
+			if (combatBoost < 50) {
+				combatBoost = combatBoost - 50;
+			} else {
+				combatBoost = Math.min(10, combatBoost / 6.5);
+			}
+
+			boosts.push(`${combatBoost.toFixed(2)}% for combat stats`);
+			duration *= 1 - combatBoost / 100;
```

Comment:
You can keep this as is (without using reduceNumByPercent), as it can be negative or positive.